cloud/shared/bin/doctor_impl

```diff
@@ -112,13 +112,24 @@ class AzureCliVersionCheck(Check):
         return self._get_cloud_provider() != "azure"
 
     def is_ok(self):
-        azure_cli_version_string = self._get_command_output([
+        azure_version_info = self._get_command_output([
             "/bin/bash",
             "-c",
-            "az --version 2> /dev/null | grep -oP '^azure-cli\s+[0-9]+\.[0-9]+\.[0-9]' | grep -oP '[0-9]+\.[0-9]+\.[0-9]'"
+            "az --version 2> /dev/null",
         ])
-
-        return azure_cli_version_string and self.version_greater_than(self.MIN_VERSION, azure_cli_version_string)
+        # Search for the azure-cli version. Sample output:
+        # azure-cli                         2.36.0
+        azure_cli_lines = [l for l in azure_version_info.split('\n') if l.startswith('azure-cli')]
+        if len(azure_cli_lines) != 1:
+            print('could not find single azure-cli entry in az --version output')
+            return False
+        azure_cli_line = azure_cli_lines[0]
+        azure_cli_version_string = azure_cli_line.split(' ')[-1]
+        if not azure_cli_version_string:
+            print(f'could not determine the azure_cli_version from the az --version output: got "{azure_cli_version_string}"')
+            return False
+
+        return self.version_greater_than(self.MIN_VERSION, azure_cli_version_string)
```

Comment:
I'm getting this error:
```
  File "/Users/kianam/dev/civiform-review/cloud/shared/bin/doctor_impl", line 188, in <module>
    if check.is_ok():
  File "/Users/kianam/dev/civiform-review/cloud/shared/bin/doctor_impl", line 132, in is_ok
    return self.version_greater_than(self.MIN_VERSION, azure_cli_version_string)
  File "/Users/kianam/dev/civiform-review/cloud/shared/bin/lib/check.py", line 49, in version_greater_than
    min_version, test_version = map(self._semver_string_to_int_array, [min_version, test_version])
  File "/Users/kianam/dev/civiform-review/cloud/shared/bin/lib/check.py", line 46, in _semver_string_to_int_array
    return [int(x) for x in semver_string.split(".")]
  File "/Users/kianam/dev/civiform-review/cloud/shared/bin/lib/check.py", line 46, in <listcomp>
    return [int(x) for x in semver_string.split(".")]
ValueError: invalid literal for int() with base 10: '*'```

Reply:
from az --version:
```
▶ az --version
azure-cli                         2.34.1 *

core                              2.34.1 *
telemetry                          1.0.6

Dependencies:
msal                              1.16.0
azure-mgmt-resource               20.0.0

Python location '/usr/local/Cellar/azure-cli/2.34.1/libexec/bin/python'
Extensions directory '/Users/kianam/.azure/cliextensions'

Python (Darwin) 3.10.2 (main, Mar 26 2022, 00:25:35) [Clang 13.0.0 (clang-1300.0.29.3)]

Legal docs and information: aka.ms/AzureCliLegal


You have 2 updates available. Consider updating your CLI installation with 'az upgrade'

Please let us know how we are doing: https://aka.ms/azureclihats
and let us know if you're interested in trying out our newest features: https://aka.ms/CLIUXstudy```

Reply:
Thanks for pointing this out. I switched back to a regexp for doing the matching, just in Python rather than bash.